File: 2025-2/sesion-de-clase-03/java/pe/edu/pucp/inf30/softprog/config/CadenaConexion.java
package pe.edu.pucp.inf30.softprog.config;

public class CadenaConexion {
    private String servidor;
    private String schema;
    private int puerto;

    public String getServidor() {
        return servidor;
    }

    public void setServidor(String servidor) {
        this.servidor = servidor;
    }

    public String getSchema() {
        return schema;
    }

    public void setSchema(String schema) {
        this.schema = schema;
    }

    public int getPuerto() {
        return puerto;
    }

    public void setPuerto(int puerto) {
        this.puerto = puerto;
    }
    
    CadenaConexion(Builder builder) {
        this.servidor = builder.getServidor();
        this.schema = builder.getSchema();
        this.puerto = builder.getPuerto();
    }

    @Override
    public String toString() {
        return String.format(
                "jdbc:mysql://%s:%d/%s?useSSL=false&"
                    + "allowPublicKeyRetrieval=true", 
                this.servidor, 
                this.puerto, 
                this.schema);
    }
    
    public static class Builder {
        private String servidor;
        private String schema;
        private int puerto;
        
        public String getServidor() {
            return this.servidor;
        }
        
        public String getSchema() {
            return this.schema;
        }
        
        public int getPuerto() {
            return this.puerto;
        }
        
        public Builder servidor(String servidor) {
            this.servidor = servidor;
            return this;
        }
        
        public Builder schema(String schema) {
            this.schema = schema;
            return this;
        }
        
        public Builder puerto(int puerto) {
            this.puerto = puerto;
            return this;
        }
        
        public CadenaConexion build() {
            return new CadenaConexion(this);
        }
    }
}

File: 2025-2/sesion-de-clase-03/dotNET/CadenaConexion.cs
/*
 * Esta es una implementacion del patron Fluent Builder
 */
namespace PUCP.Inf30.SoftProg.Config {
    public class CadenaConexion {
        public string Servidor { get; set; }
        public string Schema { get; set; }
        public int Puerto { get; set; }

        CadenaConexion(Builder builder) {
            this.Servidor = builder.Servidor;
            this.Schema = builder.Schema;
            this.Puerto = builder.Puerto;
        }

        public class Builder {
            public string Servidor { get; private set; }
            public string Schema { get; private set; }
            public int Puerto { get; private set; }

            public Builder ConServidor(string servidor) {
                this.Servidor = servidor;
                return this;
            }

            public Builder ConSchema(string schema) { 
                this.Schema = schema;
                return this;
            }

            public Builder EnPuerto(int puerto) {
                this.Puerto = puerto;
                return this;
            }

            public CadenaConexion Build() {
                return new CadenaConexion(this);
            }
        }

        public override string ToString() {
            return string.Format("version1-jdbc:mysql://{0}:{1}/{2}?useSSL=false&allowPublicKeyRetrieval=true", Servidor, Schema, Puerto);
        }
    }
}
